nightfall-client/src/event-handlers/subscribe.mjs

```diff
@@ -7,13 +7,28 @@ import { waitForContract } from '@polygon-nightfall/common-files/utils/contract.
 import constants from '@polygon-nightfall/common-files/constants/index.mjs';
 import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
 
-const { STATE_CONTRACT_NAME } = constants;
+const {
+  STATE_CONTRACT_NAME,
+  CHALLENGES_CONTRACT_NAME,
+  SHIELD_CONTRACT_NAME,
+  PROPOSERS_CONTRACT_NAME,
+} = constants;
 
 // eslint-disable-next-line import/prefer-default-export
 export async function startEventQueue(callback, ...args) {
-  const emitter = (await waitForContract(STATE_CONTRACT_NAME)).events.allEvents();
-  emitter.on('data', event => callback(event, args));
-  emitter.on('changed', event => callback(event, args));
+  const contractNames = [
+    STATE_CONTRACT_NAME,
+    SHIELD_CONTRACT_NAME,
+    CHALLENGES_CONTRACT_NAME,
+    PROPOSERS_CONTRACT_NAME,
+  ];
+  const contracts = await Promise.all(contractNames.map(c => waitForContract(c)));
+  const emitters = contracts.map(e => {
+    const emitterC = e.events.allEvents();
+    emitterC.on('changed', event => callback(event, args));
+    emitterC.on('data', event => callback(event, args));
+    return emitterC;
+  });
   logger.debug('Subscribed to layer 2 state events');
```

Comment:
Better `Subscribed to layer 2 contracts events` because now it's not only the state contract.